torch/csrc/autograd/python_variable.cpp

```diff
@@ -1436,3 +1467,116 @@ bool THPVariable_initModule(PyObject *module)
   torch::autograd::initTensorImplConversion(module);
   return true;
 }
+
+namespace {
+
+bool isPythonTensor(const Tensor& tensor) {
+  return tensor.unsafeGetTensorImpl()->key_set().has(c10::DispatchKey::Python);
+}
+
+void concrete_dispatch_fn(const c10::impl::PyInterpreter*, const c10::OperatorHandle& op, torch::jit::Stack* stack) {
+  const auto& schema = op.schema();
+  const auto num_returns = schema.returns().size();
+
+  const auto num_arguments = schema.arguments().size();
+  auto arguments = torch::jit::pop(*stack, num_arguments);
+
+  // Parse the name into namespace and name (no overload_name)
+  // TODO: put this into the library
+  const auto& qualified_name = op.operator_name().name;
+  auto pos = qualified_name.find("::");
+  TORCH_INTERNAL_ASSERT(pos != std::string::npos, qualified_name);
+  // Make me some null terminated strings
+  std::string ns_str = qualified_name.substr(0, pos);
+  const char* ns = ns_str.c_str();
+  const char* func_name = qualified_name.c_str() + pos + strlen("::");
+
+  // The plan: convert all the arguments back into PyObjects,
+  // extracting out the tensor handles, then call
+  // handle_torch_function_no_python_arg_parser
+
+  py::gil_scoped_acquire g;
+
+  std::vector<py::handle> overloaded_args;
+  auto args = py::reinterpret_steal<py::object>(PyTuple_New(num_arguments));
+  py::dict kwargs;  // TODO: actually populate kwargs sometimes?
+  // For now, overloads get coalesced.  Might be easier for users if they get
+  // overload resolution but is more complicated (need to expose separate
+  // functions per overload)
+  py::handle torch_api_function = py::module::import("torch").attr("ops").attr(ns).attr(func_name);
```

Comment:
I guess we can cache `py::module::import("torch").attr("ops")`

Reply:
I grepped `git grep py::module::import | grep static` and there are no occurrences, for what it's worth. Non leaking caches are a bit worrisome, IMO, because destruction can interact unpredictably with Python interpreter shutdown...

Reply:
Not going to do this.